src/shared/codelens/codeLensUtils.ts

```diff
@@ -0,0 +1,183 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import * as path from 'path'
+import * as vscode from 'vscode'
+import * as nls from 'vscode-nls'
+const localize = nls.loadMessageBundle()
+
+import { LambdaHandlerCandidate } from '../lambdaHandlerSearch'
+import {
+    SamCliProcessInvoker,
+    SamCliTaskInvoker
+} from '../sam/cli/samCliInvoker'
+import { SettingsConfiguration } from '../settingsConfiguration'
+
+// ------- Temporary logging utilities to be replaced by upcoming logger implementation --------------
+// TODO: Remove this code and replace usage with upcoming src/shared/logger
+type LogLevel = 'debug' | 'info' | 'warn' | 'error'
+const log = (channel: vscode.OutputChannel, level: LogLevel, msg: string, err?: Error): void => {
+    channel.show()
```

Comment:
We shouldn't force show the channel unless loglevel is `error`

Reply:
Update: 
If I start typing in a Python file, this gets triggered and moves the focus out of the editor over to the output pane.
We shouldn't _ever_ steal focus. Set `preserveFocus` to `true` on the call to `show()`

Reply:
Good feedback. Thanks. Done.
✔️